src/graph/sampler.cc

```diff
@@ -305,7 +307,7 @@ NodeFlow ConstructNodeFlow(std::vector<dgl_id_t> neighbor_list,
   }
   layer_off_data[0] = 0;
   layer_off_data[1] = layer_offsets[num_hops] - layer_offsets[num_hops - 1];
-  size_t out_layer_idx = 1;
+  int out_layer_idx = 1;
```

Comment:
why changing the types?

Reply:
It throws type comparison warnings otherwise.